domain/schema/model.go

```diff
@@ -326,7 +325,8 @@ ON application (name);
 func charmSchema() schema.Patch {
 	return schema.MakePatch(`
 CREATE TABLE charm (
-    uuid    TEXT PRIMARY KEY
+    uuid    TEXT PRIMARY KEY,
+    url     TEXT NOT NULL
```

Comment:
This doesn't seem right. We shouldn't be indexing on the URL. Also I do find it strange you can annotate a charm, but ignoring that.

Reply:
Yeah I needed a reference in this table, saw that the `CharmKind.Id()` in the `names.Tag` outputs a url, so that's why I added that here. It'll prob change when we add the charm, then whatever it is the selector in the annotation state layer for that will change, I'll add a todo for that.

Reply:
Yes, this need consideration when we move charms over. We won't be referring to them by URL. I think it will be UUID, but maybe another natural key.